### ISSUE TITLE
Should this have errored for missing return?

### ISSUE THREAD

**Issue by asottile** [python/mypy#5001]
### input

```python
import arbitrary_module

def f() -> arbitrary_module.MyType:
    x = 1

# reveal_type(arbitrary_module.MyType)
```

### output

```
$ mypy test.py --ignore-missing-imports
$ echo $?
0
```

### expected (?)

```
test.py:3: error: Missing return statement
```

### explanation

I get _why_ this is happening, just feels non-ideal to me.

The revealed type above is `Any`, and implicit `return None` matches `Any` so this is allowed.  The case I'd like to prevent is the accidental omission of a `return` statement here (_this example has been reduced from an actual production issue_).

From an implementation perspective I guess I'd want the types generated from `--ignore-missing-imports` to be non-nullable, though this might be a big change.

I'm sure there's a duplicate of this issue but I was struggling to search for it :S

**Comment by ethanhs** [comment]
I agree that I would want mypy to catch this. My gut says this shouldn't be too hard to implement.

**Comment by asottile** [comment]
Oh actually, that isn't it at all. The one-line function thing is just for trivial functions (only ellipsis, pass, or docstring). Updated the example anyway to alleviate any confusion with that

**Comment by gvanrossum** [comment]
OK, let me rewrite the topic as something more reasonable. Though this is probably a duplicate.

**Comment by JukkaL** [comment]
I think that mypy currently allows leaving out an explicit return statement if the return type is `Any`. Maybe this special case should only apply if there is an implicit `Any` return type. If there is an explicit `Any` return type, mypy could still complain about an implicit `None` return.
